lib/storage/index_db_test.go

```diff
@@ -1330,6 +1331,146 @@ func TestMatchTagFilters(t *testing.T) {
 	}
 }
 
+func TestSearchTSIDWithTimeRange(t *testing.T) {
+	metricIDCache := workingsetcache.New(1234, time.Hour)
+	metricNameCache := workingsetcache.New(1234, time.Hour)
+	defer metricIDCache.Stop()
+	defer metricNameCache.Stop()
+
+	currMetricIDs := &hourMetricIDs{
+		isFull: true,
+		m:      &uint64set.Set{},
+	}
+
+	var hmCurr atomic.Value
+	hmCurr.Store(currMetricIDs)
+
+	prevMetricIDs := &hourMetricIDs{
+		isFull: true,
+		m:      &uint64set.Set{},
+	}
+	var hmPrev atomic.Value
+	hmPrev.Store(prevMetricIDs)
+
+	dbName := "test-index-db-ts-range"
+	db, err := openIndexDB(dbName, metricIDCache, metricNameCache, &hmCurr, &hmPrev)
+	if err != nil {
+		t.Fatalf("cannot open indexDB: %s", err)
+	}
+	defer func() {
+		db.MustClose()
+		if err := os.RemoveAll(dbName); err != nil {
+			t.Fatalf("cannot remove indexDB: %s", err)
+		}
+	}()
+
+	is := db.getIndexSearch()
+	defer db.putIndexSearch(is)
+
+	// Create a bunch of per-day time series
+	const days = 5
+	const metricsPerDay = 1000
+	theDay := time.Date(2019, time.October, 15, 5, 1, 0, 0, time.UTC)
+	now := uint64(timestampFromTime(theDay))
+	currMetricIDs.hour = now / msecPerHour
+	prevMetricIDs.hour = (now - msecPerHour) / msecPerHour
+	baseDate := now / msecPerDay
+	var metricNameBuf []byte
+	for day := 0; day < days; day++ {
+		var tsids []TSID
+		for metric := 0; metric < metricsPerDay; metric++ {
+			var mn MetricName
+			mn.MetricGroup = []byte("testMetric")
+			mn.AddTag(
+				"constant",
+				"const",
+			)
+			mn.AddTag(
+				"day",
+				fmt.Sprintf("%v", day),
+			)
+			mn.AddTag(
+				"uniqueid",
+				fmt.Sprintf("%v", metric),
+			)
+			mn.sortTags()
+
+			metricNameBuf = mn.Marshal(metricNameBuf[:0])
+			var tsid TSID
+			if err := is.GetOrCreateTSIDByName(&tsid, metricNameBuf); err != nil {
+				t.Fatal(fmt.Errorf("unexpected error when creating tsid for mn:\n%s: %s", &mn, err))
+				return
+			}
+			tsids = append(tsids, tsid)
+		}
+
+		// Add the metrics to the per-day stores
+		date := baseDate - uint64(day*msecPerDay)
+		for i := range tsids {
+			tsid := &tsids[i]
+			if err := db.storeDateMetricID(date, tsid.MetricID); err != nil {
+				t.Fatal(fmt.Errorf("error in storeDateMetricID(%d, %d): %s", date, tsid.MetricID, err))
+				return
+			}
+		}
+
+		// Add the the hour metrics caches
+		if day == 0 {
+			for i := 0; i < 256; i++ {
+				prevMetricIDs.m.Add(tsids[i].MetricID)
+				currMetricIDs.m.Add(tsids[i].MetricID)
+			}
+			prevMetricIDs.iidx = newInmemoryInvertedIndex()
+			prevMetricIDs.iidx.MustUpdate(db, prevMetricIDs.m)
+			currMetricIDs.iidx = newInmemoryInvertedIndex()
+			currMetricIDs.iidx.MustUpdate(db, currMetricIDs.m)
+		}
+	}
+
+	// Flush index to disk, so it becomes visible for search
+	db.tb.DebugFlush()
+
+	// Create a filter that will match series that occur across multiple days
+	var tfs TagFilters
+	tfs.Reset()
```

Comment:
The `tfs.Reset` isn't needed here

Reply:
It did not work without the `Reset` (it's needed to set `commonPrefix` in the TagFilters), but I replaced `var tfs TagFilters` with `tfs := NewTagFilters()` and that works.